# ITMGT 25.03 EE
# Assignment 3A

This is a data-cleaning assignment and is the first of two parts. The second part (3B) will be given Monday or Tuesday next week. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.


The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called cleanrecweekdata.csv **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats.html**. The requirements for the summary statistics are described below. **(10 points)**

This entire assignment (3A) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Tuesday, July 9, 2019** at 12:00NN.

In [1]:
import csv

recweek_list = list()

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for line in csv_file:
        for row in csv_reader:
            if line_count == 0:
                pass
            else:
                recweek_list.append({"firstname":row[0], "lastname":row[1], "birthmonth":row[2], "birthday":row[3], "birthyear":row[4], "course":row[5], "phonenumber":row[6]})
            line_count += 1

recweek_list

[{'firstname': 'name2',
  'lastname': 'lastname2',
  'birthmonth': 'May',
  'birthday': '26',
  'birthyear': '2001',
  'course': 'Management',
  'phonenumber': '927 212 6276'},
 {'firstname': 'name3',
  'lastname': 'lastname3',
  'birthmonth': '5',
  'birthday': '16',
  'birthyear': '1997',
  'course': 'BS CTM',
  'phonenumber': '+63 918 253 7017'},
 {'firstname': 'name4',
  'lastname': 'lastname4',
  'birthmonth': 'Dec',
  'birthday': '9',
  'birthyear': '97',
  'course': 'BS ITE',
  'phonenumber': '918 714 4360'},
 {'firstname': 'name5',
  'lastname': 'lastname5',
  'birthmonth': 'Sep',
  'birthday': '5',
  'birthyear': '97',
  'course': 'BS CTM',
  'phonenumber': '+63 927 462 1674'},
 {'firstname': 'name6',
  'lastname': 'lastname6',
  'birthmonth': 'March',
  'birthday': '2',
  'birthyear': '1999',
  'course': 'Management',
  'phonenumber': '+63 918 629 5268'},
 {'firstname': 'name7',
  'lastname': 'lastname7',
  'birthmonth': '11',
  'birthday': '10',
  'birthyear': '98',
  'cours

In [2]:
# data cleaning
import re


# course cleaning
course_list = [c["course"] for c in recweek_list]
course_replacements = [("^Legal Management|BS LM|LM","BS Legal Management"),
                       ("Communications Technology|BS CTM|CTM","BS Communications Technology Management"),
                      ("Information Technology Entrepreneurship|BS ITE|ITE","BS IT Entrepreneurship"),
                      ("^Management Engineering|BS ME|ME","BS Management Engineering"),
                      ("^Management|BS MGT|MGT","BS Management")]

for one,two in course_replacements:
    course_list = [re.sub(one, two, course) for course in course_list]
    

# birthmonth data cleaning --working
birthmonth_list = [c["birthmonth"] for c in recweek_list]
month_replacements = [("January|Jan|1","January"),("February|Feb|2","February"),("March|Mar|3","March"),
                      ("April|Apr|4","April"),('May|5','May'),("June|Jun|6","June"),("July|Jul|7","July"),
                      ("August|Aug|8","August"),("September|Sep|9","September"),("January0|October|Oct|10","October"),
                      ("JanuaryJanuary|November|Nov|11","November"),("JanuaryFebruary|December|Dec|12","December")
                     ]

for dirty,clean in month_replacements:
    birthmonth_list = [re.sub(dirty, clean, month) for month in birthmonth_list]


# birthmonth data cleaning --> some numbers are repeating ("1199977")????
birthyear_list = [c["birthyear"] for c in recweek_list]
years_replacement = [("^97","1997"),("^98","1998"),("^99","1999")]

for three,four in years_replacement:
    birthyear_list = [re.sub(three, four, year) for year in birthyear_list]
    

# phonenumber data cleaning
phonenumber_list = [c["phonenumber"] for c in recweek_list]
phone_replacement = [("^\+63 ","0"),("^9","09")]

for five,six in phone_replacement:
    phonenumber_list = [re.sub(five,six,number) for number in phonenumber_list]

# other lists
firstname = [c["firstname"] for c in recweek_list]
lastname = [c["lastname"] for c in recweek_list]
birthday = [c["birthday"] for c in recweek_list]

# new csv file 
clean_recweek_list = list()
with open('cleanrecweekdata.csv','w+') as clean_file:
    csv_writer = csv.writer(clean_file) # creating a variable in this line allows you to create rows
    
    csv_writer.writerow(['firstname','lastname','birthmonth','birthday','birthyear','course','phonenumber'])
    
    for i in range(len(firstname)):
        csv_writer.writerow([firstname[i],lastname[i],birthmonth_list[i],birthday[i],birthyear_list[i],course_list[i],phonenumber_list[i]])
    



# Required Summary Statistics
1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

Answer these questions in an html file called stats.html. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


In [3]:
stats = """
    <html>
       <head>
            <style>
                table, th, td {
                  border: 1px solid black;
                  font-family: verdana;
                }
                th, td {
                  padding: 5px;
                  text-align: center;    
                }
                h2 {
                  font-family: verdana;
                  font-size: 200%;
                  font-weight: bold;
                  text-align: center;
                }
                
            </style>
        </head>
        <body>
             <h2>SUMMARY STATISTICS</h2>
             <table style="width:100%">
                 <tr>
                     <th style="font-size: 100%">STATISTIC</th>
                     <th style="font-size: 100%">VALUES</th>
                 </tr>
                 
                 <tr>
                     <td style="font-size: 100%">People born in each month</td>
                         <td>
                         <table style="width:100%; border-collapse: collapse;">
                         <th>MONTH</th>
                         <th>COUNT</th>
                         <tr>
                             <td><b>January</b></td>
                             <td>270</td>
                         </tr>
                         <tr>
                             <td><b>February</b></td>
                             <td>232</td>
                         </tr>
                         <tr>
                             <td><b>March</b></td>
                             <td>246</td>
                         </tr>
                         <tr>
                             <td><b>April</b></td>
                             <td>267</td>
                         </tr>
                         <tr>
                             <td><b>May</b></td>
                             <td>273</td>
                         </tr>
                         <tr>
                             <td><b>June</b></td>
                             <td>259</td>
                         </tr>
                         <tr>
                             <td><b>July</b></td>
                             <td>249</td>
                         </tr>
                         <tr>
                             <td><b>August</b></td>
                             <td>241</td>
                         </tr>
                         <tr>
                             <td><b>September</b></td>
                             <td>232</td>
                         </tr>
                         <tr>
                             <td><b>October</b></td>
                             <td>178</td>
                         </tr>
                         <tr>
                             <td><b>November</b></td>
                             <td>277</td>
                         </tr>
                         <tr>
                             <td><b>December</b></td>
                             <td>276</td>
                         </tr>
                         </table>
                         </td>
                 </tr>
                 
                 <tr>
                     <td>People born on each day</td>
                         <td>
                         <table style="width:100%; border-collapse: collapse;"">
                         <th>DATE</th>
                         <th>COUNT</th>
                         <tr>
                             <td><b>1</b></td>
                             <td>112</td>
                         </tr>
                         <tr>
                             <td><b>2</b></td>
                             <td>106</td>
                         </tr>
                         <tr>
                             <td><b>3</b></td>
                             <td>90</td>
                         </tr>
                         <tr>
                             <td><b>4</b></td>
                             <td>110</td>
                         </tr>
                         <tr>
                             <td><b>5</b></td>
                             <td>100</td>
                         </tr>
                         <tr>
                             <td><b>6</b></td>
                             <td>99</td>
                         </tr>
                         <tr>
                             <td><b>7</b></td>
                             <td>88</td>
                         </tr>
                         <tr>
                             <td><b>8</b></td>
                             <td>111</td>
                         </tr>
                         <tr>
                             <td><b>9</b></td>
                             <td>97</td>
                         </tr>
                         <tr>
                             <td><b>10</b></td>
                             <td>109</td>
                         </tr>
                         <tr>
                             <td><b>11</b></td>
                             <td>105</td>
                         </tr>
                         <tr>
                             <td><b>12</b></td>
                             <td>98</td>
                         </tr>
                         <tr>
                             <td><b>13</b></td>
                             <td>119</td>
                         </tr>
                         <tr>
                             <td><b>14</b></td>
                             <td>101</td>
                         </tr>
                         <tr>
                             <td><b>15</b></td>
                             <td>121</td>
                         </tr>
                         <tr>
                             <td><b>16</b></td>
                             <td>114</td>
                         </tr>
                         <tr>
                             <td><b>17</b></td>
                             <td>116</td>
                         </tr>
                         <tr>
                             <td><b>18</b></td>
                             <td>107</td>
                         </tr>
                         <tr>
                             <td><b>19</b></td>
                             <td>113</td>
                         </tr>
                         <tr>
                             <td><b>20</b></td>
                             <td>90</td>
                         </tr>
                         <tr>
                             <td><b>21</b></td>
                             <td>123</td>
                         </tr>
                         <tr>
                             <td><b>22</b></td>
                             <td>103</td>
                         </tr>
                         <tr>
                             <td><b>23</b></td>
                             <td>116</td>
                         </tr>
                         <tr>
                             <td><b>24</b></td>
                             <td>108</td>
                         </tr>
                         <tr>
                             <td><b>25</b></td>
                             <td>116</td>
                         </tr>
                         <tr>
                             <td><b>26</b></td>
                             <td>114</td>
                         </tr>
                         <tr>
                             <td><b>27</b></td>
                             <td>102</td>
                         </tr>
                         <tr>
                             <td><b>28</b></td>
                             <td>101</td>
                         </tr>
                         <tr>
                             <td><b>29</b></td>
                             <td>0</td>
                         </tr>
                         <tr>
                             <td><b>30</b></td>
                             <td>0</td>
                         </tr>
                         <tr>
                             <td><b>31</b></td>
                             <td>0</td>
                         </tr>
                         
                         </table>
                         </td>
                    
                 </tr>

                 <tr>
                     <td>People born in each year</td>
                         <td>
                         <table style="width:100%; border-collapse: collapse;"">
                         <th>YEAR</th>
                         <th>COUNT</th>
                         <tr>
                             <td><b>1997</b></td>
                             <td>757</td>
                         </tr>
                         <tr>
                             <td><b>1998</b></td>
                             <td>746</td>
                         </tr>
                         <tr>
                             <td><b>1999</b></td>
                             <td>756</td>
                         </tr>
                         <tr>
                             <td><b>2000</b></td>
                             <td>380</td>
                         </tr>
                         <tr>
                             <td><b>2001</b></td>
                             <td>362</td>
                         </tr>
                         
                         </table>
                         </td>
                 </tr>
                
                 <tr>
                     <td>People coming from each course</td>
                         <td>
                         <table style="width:100%; border-collapse: collapse;"">
                         <th>COURSE</th>
                         <th>COUNT</th>
                         
                         <tr>
                             <td><b>BS Legal Management</b></td>
                             <td>585</td>
                         </tr>
                         <tr>
                             <td><b>BS IT Entrepreneurship</b></td>
                             <td>590</td>
                         </tr>
                         <tr>
                             <td><b>BS Management</b></td>
                             <td>629</td>
                         </tr>
                         <tr>
                             <td><b>BS Management Engineering</b></td>
                             <td>607</td>
                         </tr>
                         <tr>
                             <td><b>BS Communications Technology Management</b></td>
                             <td>589</td>
                         </tr>
                         </table>
                         </td>
                 </tr>
                 
                 <tr>
                     <td>People having numbers from Globe and Smart</td>
                         <td>
                         <table style="width:100%; border-collapse: collapse;"">
                         <th>COURIER</th>
                         <th>COUNT</th>
                         <tr>
                             <td><b>Smart</b></td>
                             <td>955</td>
                         </tr>
                         <tr>
                             <td><b>Globe</b></td>
                             <td>2045</td>
                         </tr>
                         </table>
                         </td>
                 </tr>
             </table>
        </body>
    </html>



"""

with open('stats.html',"w+") as stats_file:
    stats_file.write(stats)